df_engine_extensions/src/dist_sql_query/resolver.rs

```diff
@@ -58,6 +58,22 @@ impl Resolver {
     pub fn resolve_partitioned_scan(
         &self,
         plan: Arc<dyn ExecutionPlan>,
+    ) -> DfResult<Arc<dyn ExecutionPlan>> {
+        let resolved_plan = self.resolve_partitioned_scan_internal(plan)?;
+
+        if let Some(plan) = resolved_plan
+            .as_any()
+            .downcast_ref::<ResolvedPartitionedScan>()
+        {
+            Ok(plan.push_down_finished())
+        } else {
+            Ok(resolved_plan)
+        }
+    }
+
+    pub fn resolve_partitioned_scan_internal(
```

Comment:
Add some comments explain how it will translate plan.

Give a demo input plan, and the resolved output plan.